### ISSUE TITLE
Accessibility WAIG 2.1 conformance level A

### ISSUE THREAD

**Issue by MichaelMelena** [havit/Havit.Blazor#317]
List of changes that once completed will allow developers to create content that is in accordance with [Web Content Accessibility Guidelines 2.1](https://www.w3.org/TR/WCAG21/) at conformance level A

- HxFormValue 
    * [x] 2.1.1 is not focusable and cannot be accessed by arrow keys
   
- HxInputDate
    * [x] 2.1.1 context menu cannot be access by TAB
    * [ ] 2.1.1 Can't select date from the dropdown via keyboard (TAB + Arrow keys)
    * [x] 4.1.2 context button doesn't have name

- HxMultiSelect
    * [x] 1.3.1 The HxMultiSelect has the `listbox` role but possible options have the required role `option` . 
    ![image](https://user-images.githubusercontent.com/24386972/186443431-6c797847-14dd-4aee-9fab-ad6f1fdcb098.png)

- HxIcon
    * [x] 1.1.1 icons don't have alternative text. We should specify title, aria-lable or allow the element to be hidden from assistive technologies because it servers only decorative purpose.

- HxChip
    * [x] 4.1.2 close icon doesn't have any accessible name. Add `aria-label` to it.
    ![image](https://user-images.githubusercontent.com/24386972/186442469-12c0bc1b-718b-40e3-ac47-4b12c3b324e1.png)

- HxSpinner
    * [x] 2.2.2 stop the spinner animatio by using [prefers-reduced-motion](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion) CSS media query

- HxPlaceholder
    * [ ] 2.2.2 stop the placeholder animation by using [prefers-reduced-motion](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion) CSS media query
   
- HxTreeView
    * [ ] 2.1.1 cannot be controlled by keyboard at all.
        * [ ] You should be able to navigate between items using **Tab** or **Arrow keys**. 
        * [ ] Pressing **spacebar** or **enter** on folder should toggle the folder.  
        * [ ] Pressing Right arrow key on folder should open the folder and  pressing Left arrow key should close the folder.
        * [ ] Pressing tab on folder should focus the plus icon

- HxInputFileDropZone
    * Is fully keyboard accessible ✅ 
    * [x] 1.4.1 Highlight the drop zone area when the underlying input has focus. 

 - HxAutoSuggest
    * [x] 2.1.1 Cannot select clear text button using a keyboard. Should be able to switch to it by pressing a Tab. ![image](https://user-images.githubusercontent.com/24386972/193053285-e00e7c51-efe6-4675-8725-28e73f7eaced.png)

- HxCalendar
    * [x] 2.1.1 Cannot select date using a keyboard (you can only focus the arrows that change the month). You should be able to switch to the days section by pressing a Tab
 ![image](https://user-images.githubusercontent.com/24386972/193054551-f9494738-4b2c-4a19-8453-8a4929754479.png)

- HxIpnutDate
    * Inherits all the keyboard problems from `HxCalendar`
    * [ ] 2.1.1 You can't navigate to calendar window by a keyboard (it show only after a left mouse click inside the input)
    * [x] 2.1.1 When you open the calendar you cannot navigate to the **Clear** and **Today** buttons. When you press Tab after the right arrow at the top of the calendar the calendar window closes.
![image](https://user-images.githubusercontent.com/24386972/193056173-dadf4246-071d-45ab-ac0c-c57caafbeaf0.png)

- HxInputDateRange
    *  Inherits all the keyboard problems from `HxCalendar` and `HxInputDate`
    *  ❗ Unlike in `HxInputDate` here we can navigate to **Clear** button using a Tab when the calendar window is already open.

- HxInputTag
    * [x] 2.1.1 Cannot "trigger" delete icon on individual tags with keyboard
![image](https://user-images.githubusercontent.com/24386972/193057659-4dbb3d94-95d6-4430-8b29-0768e2e1a667.png)

**Comment by hakenr** [comment]
I would like to discuss individual "task" one by one and agree on implementation.
Please do not start the implementation without that.

**Comment by hakenr** [comment]
HxInputTags - cannot "trigger" delete icon on individual tags with keyboard
![image](https://user-images.githubusercontent.com/12828888/189112603-b797acd5-4b82-498b-a76c-b5ae57c51d96.png)

**Comment by crdo** [comment]
@MichaelMelena the context menu in HxInputDate has been removed in the last major version and so I have marked the related tasks as complete :)

**Comment by MichaelMelena** [comment]
> HxInputTags - cannot "trigger" delete icon on individual tags with keyboard ![image](https://user-images.githubusercontent.com/12828888/189112603-b797acd5-4b82-498b-a76c-b5ae57c51d96.png)

Added to the list in the first comment

**Comment by crdo** [comment]
@hakenr I have implemented most of the low hanging fruits today.

I need to discuss your opinion on `HxInputDate`, where I understand that currently we say that the user can have a full control over the input from keyboard w/o displaying the calendar (it is currently not displayed on focus), but that does not satisfy cases like reservations or flight booking where you need to see availability of dates.

Another bigger stone is `HxTreeView`, where the keyboard controls are not supported in Bootstrap either and by the [docs](https://getbootstrap.com/docs/5.3/components/collapse/#accessibility) it needs to be implemented separately.

Cc @alexax578 @MichaelMelena
